Sync-EntraID-Groupmemberships-To-HelloID-Selfservice-Productassignments.ps1

```diff
@@ -1,47 +1,52 @@
+}
+$informationPreference = "Continue"
+$WarningPreference = "Continue"
+
+# Set TLS to accept TLS, TLS 1.1 and TLS 1.2
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls -bor [Net.SecurityProtocolType]::Tls11 -bor [Net.SecurityProtocolType]::Tls12
+
 # Make sure to create the Global variables defined below in HelloID
 #HelloID Connection Configuration
 # $script:PortalBaseUrl = "" # Set from Global Variable
 # $portalApiKey ="" # Set from Global Variable
 # $portalApiSecret = "" # Set from Global Variable
 
-#AzureAD Connection Configuration
+#EntraID Connection Configuration
 $MSGraphBaseUri = "https://graph.microsoft.com/" # Fixed value
-# $AzureADtenantID = "" # Set from Global Variable
-# $AzureADAppId = "" # Set from Global Variable
-# $AzureADAppSecret = "" # Set from Global Variable
-$AzureADGroupsSearchFilter = "`$search=`"displayName:department_`"" # Optional, when no filter is provided ($AzureADGroupsSearchFilter = $null), all groups will be queried - Only displayName and description are supported with the search filter. Reference: https://learn.microsoft.com/en-us/graph/search-query-parameter?tabs=http#using-search-on-directory-object-collections
+# $EntraTenantId = "" # Set from Global Variable
+# $EntraAppID = "" # Set from Global Variable
+# $EntraAppSecret = "" # Set from Global Variable
+
+$entraIDGroupsSearchFilter = "`$search=`"displayName:department_`"" # Optional, when no filter is provided ($entraIDGroupsSearchFilter = $null), all groups will be queried - Only displayName and description are supported with the search filter. Reference: https://learn.microsoft.com/en-us/graph/search-query-parameter?tabs=http#using-search-on-directory-object-collections
 
 #HelloID Self service Product Configuration
-$ProductSkuPrefix = 'AADGRP' # Optional, when no SkuPrefix is provided ($ProductSkuPrefix = $null), all products will be queried
-$PowerShellActionName = "Add-AzureADUserToAzureADGroup" # Define the name of the PowerShell action
+$ProductSkuPrefix = 'ENTRAGRP' # Optional, when no SkuPrefix is provided ($ProductSkuPrefix = $null), all products will be queried
+$PowerShellActionName = "Add-EntraIDUserToEntraIDGroup" # Define the name of the PowerShell action
 
 #Correlation Configuration
 # The name of the property of HelloID Self service Product action variables to match to AD Groups (name of the variable of the PowerShell action that contains the group)
 $PowerShellActionVariableCorrelationProperty = "GroupId"
 # The name of the property of AD groups to match Groups in HelloID Self service Product actions (the group)
-$azureADGroupCorrelationProperty = "id"
-# The name of the property of AD users to match to HelloID users
-$azureADUserCorrelationProperty = "userPrincipalName"
-# The name of the property of HelloID users to match to AD users
+$entraIDGroupCorrelationProperty = "id"
+# The name of the property of Entra ID users to match to HelloID users
+$entraIDUserCorrelationProperty = "id"
```

Comment:
klopt dit wel? in de oude variant was die userPrincipalName

Reply:
Ik heb hem nu getest bij mij zelf en de klant met een AzureAD sync. Mogelijk is het wel verstandiger om userPrinicpalName in te laten staan. Zal ik nog even testen aan aanpassen

Reply:
Het volgende werkend getest:
$entraIDUserCorrelationProperty = "userPrincipalName"
$helloIDUserCorrelationProperty = "userName"

Of
$entraIDUserCorrelationProperty = "id"
$helloIDUserCorrelationProperty = "immutableId"

Beide is met "AzureAD" sync.

Met AD verwacht ik dat onderstaande zou moeten werken (niet getest):
$entraIDUserCorrelationProperty = "userPrincipalName"
$helloIDUserCorrelationProperty = "userAttributes_userprincipalname"

Kwestie van users uitlezen uit de API en properties aan elkaar linken. Zal de tekst #note nog iets uitbreiden